test/OpenTelemetry.Exporter.Geneva.Tests/OtlpProtobufMetricExporterTests.cs

```diff
@@ -891,6 +891,82 @@ public void GaugeSerializationMultipleMetricPoints(string instrumentName, long[]
         }
     }
 
+    [Fact]
+    public void CounterSerializationSingleMetricPoint_MultipleDataTypes()
+    {
+        using var meter = new Meter(nameof(this.CounterSerializationSingleMetricPoint_MultipleDataTypes), "0.0.1");
+
+        var exportedItems = new List<Metric>();
+        using var inMemoryReader = new BaseExportingMetricReader(new InMemoryExporter<Metric>(exportedItems))
+        {
+            TemporalityPreference = MetricReaderTemporalityPreference.Delta,
+        };
+
+        using var meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddMeter(nameof(this.CounterSerializationSingleMetricPoint_MultipleDataTypes))
+            .AddReader(inMemoryReader)
+            .Build();
+
+        sbyte sByteValue = sbyte.MaxValue;
+        sbyte negativeSbyteValue = sbyte.MinValue;
+        int intValue = 29;
+        uint uintValue = uint.MaxValue;
+        int negativeIntValue = -29;
+        short shortValue = short.MaxValue;
+        ushort ushortValue = ushort.MaxValue;
+        short negativeShortValue = -12;
+        ulong ulongValue = 1234;
+
+        TagList tagList = default;
+
+        // Keep the keys in sorted order, Sdk outputs them in sorted order.
+        tagList.Add(new("intKey", intValue));
+        tagList.Add(new("negativeByteKey", negativeSbyteValue));
+        tagList.Add(new("negativeIntKey", negativeIntValue));
+        tagList.Add(new("negativeShortKey", negativeShortValue));
+        tagList.Add(new("sByteKey", sByteValue));
+        tagList.Add(new("shortKey", shortValue));
+        tagList.Add(new("uintKey", uintValue));
+        tagList.Add(new("ulongKey", ulongValue));
+        tagList.Add(new("ushortKey", ushortValue));
```

Comment:
It's probably better to check this in the existing unit tests rather than creating a new one for each instrument. You could create and initialize `TagList` as a class field and then emit a measurement with this `TagList` in the existing unit tests of all the instruments.

Reply:
added